src/snowflake/snowpark/_internal/compiler/utils.py

```diff
@@ -193,6 +193,9 @@ def update_resolvable_node(
                           |
                         JOIN
     resolve_node(SelectSnowflakePlan, query_generator) will resolve both SelectSnowflakePlan and SnowflakePlan nodes.
+
+    This operation also resets the previously calculated cumulative_node_complexity for the node so a
```

Comment:
If we add "reset" operation in this function, this function name is not informational enough - so how about call it `update_resolvable_node_and_reset_complexity`?

Reply:
i think we also reset other recalculable fields like _sql_query, _snowflake_plan, so this name may not be accurate also. maybe we can update the comment to mention that the update includes two parts:
1) Re-Resolve the proper fields based on the child
2) Clear up re-calculatable fields like _sql_query, _snowflake_plan, _cumulative_node_complexity ect